### ISSUE TITLE
Doc category menu dropdown

### ISSUE THREAD

**Issue by jonnilundy** [electricjs/electric#44]
Right now, the only way to open a specific documentation page is by first opening the first page of that category by clicking on the category itself. I think it would provide a better UX if, by clicking on the category, it expanded and you then selected the doc page you want.

This is especially cumbersome when using it on mobile. It forces the user to tap at least 4 times (twice to open the menu, once to open the category, and once on the exact page).  Also, the first page reload to the default category page is much more noticeable on mobile.

**Comment by jbalsas** [comment]
Hey @Robert-Frampton, @marcoscv-work, didn't we implement this for the [Lexicon Site](http://hosting.lexicon-site.wedeploy.io/)?

**Comment by Robert-Frampton** [comment]
Yeah we implemented a [custom navigation](https://github.com/liferay/lexicon-site/blob/master/src/components/MainNavigation.js) component for the lexicon site.

Since it's hard to know exactly what kind of navigation is needed for the various sites that use electric, we should either...

1) Not do anything, and just let the site maintainers take control over their navigation or...

2) Make the ElectricNavigation component configurable so that you can turn this kind of functionality on or off.

My biggest concern with option 2 is that we have to maintain more, and field more issues such as this rather than just letting people make what they need.

**Comment by jonnilundy** [comment]
I agree with your concern about maintaining. I'm still wondering though if the functionality is worth the extra work. I can think of two main use cases and how they would affect the user's desire for this functionality:

1. A documentation library that is relatively small and only uses the parent items as links to the documentation pages (example: http://electricjs.com). In this case, they have no need for this component. 
2. A more extensive documentation library that pushes the limits of a two-level doc structure (example: http://wedeploy.com). In this case, I would assume that this component would almost always be preferred (especially because of the mobile implications).

Maybe this is a leap and not enough users want this functionality. My concern is that if this is not "out-of-the-box", more people will just settle for the way the docs are, rather than being provided with one that works well for their users. I especially think of the group of QA users that are moving some of their documentation to Electric right now. My assumption is that they will not try to build this component on their own. 

Again, I could totally be assuming too much. I'm curious to hear your thoughts.

**Comment by Robert-Frampton** [comment]
If we think it'll be that common, then I think it's worth baking the functionality in. We can just add a parameter that will toggle that behavior to the [ElectricNavigation](https://github.com/Robert-Frampton/electric-base-components/blob/master/src/ElectricNavigation.js) component.

Can you confirm if the navigation on http://lexicon-site.wedeploy.io/ is what you are looking for?

**Comment by jonnilundy** [comment]
Let me do some more research to see if that truly is the desired functionality to others. 
Side Note: Another thing I like about the custom component on the lexicon site is that it makes third-level navigation feel very natural as well.  

![image](https://cloud.githubusercontent.com/assets/23219848/25253795/3a714592-25d7-11e7-92dd-972929ed639b.png)
